server/app/api.py

```diff
@@ -1180,35 +1221,157 @@ def post(self, user, data):
 
 
 class SearchAPI(APIResource):
+    
+    contains_entities = False
 
     methods = {
         'index': {
             'methods': {'GET'},
             'web_args': {
                 'query': Arg(str, required=True),
-                'page': Arg(int, required=True),
-                'num_per_page': Arg(int, required=True)
+                'page': Arg(int, default=1),
+                'num_per_page': Arg(int, default=10)
+            }
+        },
+        'download': {
+            'methods': {'GET'},
+            'web_args': {
+                'query': Arg(str, required=True),
+                'page': Arg(int, default=1),
+                'num_per_page': Arg(int, default=10),
+                'all': Arg(bool, default=False)
             }
         }
     }
+    
+    defaults = {
+        'onlywcode': (op.__eq__, 'True')
+    }
+    
+    operators = {
+        'eq': op.__eq__,
+        'equal': op.__eq__,
+        'lt': op.__lt__,
+        'gt': op.__gt__,
+        'before': op.__lt__,
+        'after': op.__gt__
+    }
+    
+    flags = {
```

Comment:
What does the flags dictionary map onto? (You can either call it flags_to_X or toss it in a comment somewhere) It looks like flags goes from the name to the corresponding object.